read.go

```diff
@@ -178,9 +177,25 @@ func (r *reader) readIndexData(toc *indexTOC) (*indexData, error) {
 	d.docSectionsStart = toc.fileSections.data.off
 	d.docSectionsIndex = toc.fileSections.relativeIndex()
 
-	if readSymbols {
-		d.symIndex = toc.symbolMap.relativeIndex()
-		d.symKindIndex = toc.symbolKindMap.relativeIndex()
+	if d.metaData.IndexFormatVersion == 16 {
+		d.symbols.symKindIndex = toc.symbolKindMap.relativeIndex()
+		d.fileEndSymbol, err = readSectionU32(d.file, toc.fileEndSymbol)
+		if err != nil {
+			return nil, err
+		}
+
+		// Call readSectionBlob on each section key, and store the result in
+		// the blob value.
+		for sect, blob := range map[simpleSection]*[]byte{
```

Comment:
I'm guessing this won't be the case, but is there any possibility of two simpleSections have the same value? The downside of using a map is keys are unique.

Reply:
I don't think so. Simple sections should refer to unique locations (byte offset + sz) in the index format.

Reply:
A map is probably slower than a slice here BTW. I am speculating but maps usually require many more allocations and computations and I don't imagine the Go compiler can optimize this down to something simpler.

That said, if this isn't a critical section of code it doesn't matter.

Reply:
I can switch to a slice. Upstream Zoekt has a similar piece of code further down this function.
```go
for sect, dest := range map[simpleSection]*[]uint32{
	toc.subRepos:        &d.subRepos,
	toc.runeOffsets:     &d.runeOffsets,
	toc.nameRuneOffsets: &d.fileNameRuneOffsets,
	toc.nameEndRunes:    &d.fileNameEndRunes,
	toc.fileEndRunes:    &d.fileEndRunes,
} {
	if blob, err := d.readSectionBlob(sect); err != nil {
		return nil, err
	} else {
		*dest = fromSizedDeltas(blob, nil)
	}
}
```
This reading logic happens once when a repository is loaded in. This would affect the speed at which the indexed-search pod's `zoekt-webserver` container becomes ready.

Reply:
Ahh, got it, probably not a huge perf benefit then.